dev/cmd/dev.py

```diff
@@ -1,40 +1,25 @@
-import argparse
-import os
-import sys
-
 import spack
 import spack.cmd
-import llnl.util.tty as tty
 
-SPACKDEV_FILE = os.path.realpath(os.path.expanduser(__file__))
-SPACKDEV_LIB = os.path.realpath(os.path.join(os.path.dirname(SPACKDEV_FILE),
-                                             '..', '..', 'lib'))
-sys.path.insert(0, SPACKDEV_LIB)
+# Import sub-commands
+from ..subcommands import build_env, findext, getdeps, info, init, stage
```

Comment:
@alalazo I spoke with @chissg today and I think the thing I missed was that he didn't want to have to refer to the commands explicitly: the idea is that he could add more subcommands and those could be imported here without updating this code. Perhaps the way to reconcile this with https://github.com/spack/spack/pull/11145 is to update the commands in that PR to do the appropriate `sys.path` manipulations automatically.

The other changes in this PR (in the other files) relating to following the directory structure conventions that currently exist in Spack command extensions are agreeable.

Reply:
@scheibelp @scheibelp If the other changes are agreeable and don't prevent functionality to be implemented in SpackDev, this PR shows that we can reduce `dev.py` from 41 lines to 26 lines without a single line change in Spack. That brings me to the following two questions:

1. How many subcommands do we need to add before editing the two lines above becomes a *real* maintenance issue?
2. Is it worth to modify ~280 lines in Spack (with a ~100 lines addition) and add another 15 here just to avoid writing the import above?

Answering these two questions led to my latest review of spack/spack#11145.

Reply:
> Perhaps the way to reconcile this with spack/spack#11145 is to update the commands in that PR to do the appropriate sys.path manipulations automatically.

The current mechanism carefully crafts modules with the appropriate `__path__` and adds them to `sys.modules` to give extensions the best possible isolation. This is done [here](https://github.com/spack/spack/blob/43b18dada4442f688f2a3ef6629671cd9fd866a8/lib/spack/spack/extensions.py#L60-L78). How is adding paths to `sys.path` improving encapsulation over the current state when the new mechanism resorts to using a global variable that potentially affects every import statement in the program?

Reply:
@alalazo @scheibelp Please see https://github.com/spack/spack/pull/11145#issuecomment-548985399.

Reply:
@chissg I'll read through the comment you linked, thank you. I'd appreciate to know which answers you would give to the questions I posted above.